source/mir/ndslice/selection.d

```diff
@@ -1697,3 +1697,111 @@ struct IotaMap
         return index;
     }
 }
+
+/**
+Element-wise iteration over a slice.
+In contrast to byElement, it only returns a bidirectional forward range access.
+However it is faster for such operations.
+*/
+auto byElementIterator(size_t N, Range)(auto ref Slice!(N, Range) slice)
```

Comment:
This works only for completely dense slices in normal form. This will not work after iteration operators. If slice has completely dense normal form byElement can return an array.